src/service.py

```diff
@@ -114,43 +127,64 @@ def restart(self) -> None:
             ServiceRestartError: when restarting the service fails
         """
         # Render template and write to appropriate file if only credentials are set
-        if credentials := self.state.agent_relation_credentials:
-            with open("templates/jenkins_agent_env.conf.j2", "r", encoding="utf-8") as file:
-                template = Template(file.read())
-            # fetch credentials and set them as environments
-            environments = {
-                "JENKINS_TOKEN": credentials.secret,
-                "JENKINS_URL": credentials.address,
-                "JENKINS_AGENT": self.state.agent_meta.name,
-            }
-            # render template
-            rendered = template.render(environments=environments)
-            # Ensure that service conf directory exist
-            config_dir = Path(SYSTEMD_SERVICE_CONF_DIR)
-            config_dir.mkdir(parents=True, exist_ok=True)
-            # Write the conf file
-            logger.info("Rendering agent configuration")
-            logger.debug("%s", environments)
-            # file name (override.conf) is important for the service to import envvars
-            self._render_file(f"{config_dir.resolve().as_posix()}/override.conf", rendered, 0o644)
+        credentials = self.state.agent_relation_credentials
+        if not credentials:
+            raise ServiceRestartError("Error starting the agent service: missing configuration")
+
+        with open("templates/jenkins_agent_env.conf.j2", "r", encoding="utf-8") as file:
+            template = Template(file.read())
+        # fetch credentials and set them as environments
+        environments = {
+            "JENKINS_TOKEN": credentials.secret,
+            "JENKINS_URL": credentials.address,
+            "JENKINS_AGENT": self.state.agent_meta.name,
+        }
+        # render template file
+        rendered = template.render(environments=environments)
+        # Ensure that service conf directory exist
+        config_dir = Path(SYSTEMD_SERVICE_CONF_DIR)
+        config_dir.mkdir(parents=True, exist_ok=True)
+        # Write the conf file
+        logger.info("Rendering agent configuration")
+        logger.debug("%s", environments)
+        config_file = Path(f"{SYSTEMD_SERVICE_CONF_DIR}/override.conf")
         try:
+            self._render_file(config_file, rendered, 0o644)
```

Comment:
Does this need to be in the try/except block ?

Reply:
I think we need to put it here since this method raises `FileRenderError` which is handled in an except block a few lines below